personalize.admin.inc

```diff
@@ -152,13 +152,16 @@ function personalize_status_change_form_submit(&$form, &$form_state) {
     // Redirect to a confirm form as this has more serious implications than
     // other status changes.
     $form_state['redirect'] = array('admin/structure/personalize/manage/'. $agent_name .'/complete');
+    return;
   }
-  else {
-    // Otherwise just change the status as requested.
-    $status_map = personalize_get_agent_status_map();
-    if (personalize_agent_set_status($agent_name, $status)) {
-      drupal_set_message(t('The personalization status has been set to @status', array('@status' => $status_map[$status])));
-    }
+  // Otherwise just change the status as requested.
+  if (!personalize_agent_set_status($agent_name, $status)) {
+    return;
+  }
+  $status_map = personalize_get_agent_status_map();
+  drupal_set_message(t('The personalization status has been set to @status', array('@status' => $status_map[$status])));
+  if (current_path() === 'admin/structure/personalize') {
+    $form_state['redirect'] = array('admin/structure/personalize', array('fragment' => 'personalization-group-' . $status));
   }
```

Comment:
Good to know that difference. [Done](https://github.com/acquia/personalize_module/commit/f0b2be193e8408092a4b74e36db2d3c090091047).